client/package.json

```diff
@@ -1,7 +1,7 @@
 {
   "private": true,
   "scripts": {
-    "build": "vite build",
+    "build": "tsc && vite build",
```

Comment:
why this

Reply:
Vite does not perform type checking

Reply:
hmm change it to `tsc --noEmit` so it doesn't output the files 
but maybe we should just use this https://github.com/fi3ework/vite-plugin-checker to get TS errors in the browser on dev server